KubeArmor/feeder/feeder.go

```diff
@@ -322,6 +322,23 @@ func NewFeeder(node *tp.Node) *Feeder {
 	}
 	fd.Listener = listener
 
+	if cfg.GlobalCfg.GRPC == "0" {
```

Comment:
Won't this logic be needed before we create the listener?

Reply:
So this part is actually is used to store the port number in the pid file. The port number itself can be retrieved after we create the listener. So if we decide to listen to port 0 on the listener, net.Listen("tcp", "0"), it will actually automatically find the random available port. If this part is implemented before the listener creation, we can't get the port number

Reply:
After listening to port 0, net.Listen("tcp","0"), we then can get the random port number that the listener found by using `listener.Addr().(*net.TCPAddr).Port `